lib/web_ui/lib/src/engine/navigation/url_strategy.dart

```diff
@@ -150,15 +102,17 @@ class HashUrlStrategy extends UrlStrategy {
 
 /// Wraps a custom implementation of [UrlStrategy] that was previously converted
 /// to a [JsUrlStrategy].
-class CustomUrlStrategy extends UrlStrategy {
+class CustomUrlStrategy extends ui_web.UrlStrategy {
   /// Wraps the [delegate] in a [CustomUrlStrategy] instance.
   CustomUrlStrategy.fromJs(this.delegate);
 
   final JsUrlStrategy delegate;
 
   @override
-  ui.VoidCallback addPopStateListener(DomEventListener fn) =>
-      delegate.addPopStateListener(allowInterop(fn));
+  ui.VoidCallback addPopStateListener(ui_web.PopStateListener fn) =>
```

Comment:
`HashUrlStrategy` (above) also needs a similar change. I'm surprised the analyzer isn't screaming at you since the method has a different signature now.

Reply:
Good call!